packages/ihe-gateway-sdk/src/client/ihe-gateway.ts

```diff
@@ -1,41 +1,128 @@
-import { Patient } from "@medplum/fhirtypes";
+import axios, { AxiosInstance } from "axios";
 import { v4 as uuidv4 } from "uuid";
-import { NPIStringArray, npiStringArraySchema } from "../models/shared";
-import { XCPDGateways, XCPDRequest, xcpdGatewaysSchema } from "../models/xcpd-request";
+import { npiStringArraySchema } from "../models/shared";
+import { XCPDPayload, XCPDRequest, xcpdGatewaysSchema, XCPDResponse } from "../models/xcpd";
+import { XCA_ITI_38Request } from "../models/xca_iti_38";
+import { XCA_ITI_39Request } from "../models/xca_iti_39";
+
+const DEFAULT_AXIOS_TIMEOUT_SECONDS = 120;
+
+export enum APIMode {
+  dev = "dev",
+  integration = "integration",
+  production = "production",
+}
 
 export class IHEGateway {
+  static productionUrl = "https://ihe.metriport.com";
+  static integrationUrl = "https://ihe.staging.metriport.com";
+  static devUrl = "https://localhost:8082";
+
+  static XCPD_ENDPOINT = "/xcpd";
+  static XCA_ITI_38_ENDPOINT = "/xcadq";
+  static XCA_ITI_39_ENDPOINT = "/xcadr";
+
+  readonly api: AxiosInstance;
+  constructor(apiMode: APIMode, options: { timeout?: number } = {}) {
+    this.api = axios.create({
+      timeout: options?.timeout ?? DEFAULT_AXIOS_TIMEOUT_SECONDS * 1_000,
+      baseURL:
+        apiMode === APIMode.production
+          ? IHEGateway.productionUrl
+          : apiMode === APIMode.integration
+          ? IHEGateway.integrationUrl
+          : IHEGateway.devUrl,
+    });
+  }
+
+  /**
+   * Patient Discovery (XCPD ITI-55) request.
+   * https://profiles.ihe.net/ITI/TF/Volume2/ITI-55.html
+   *
+   * @param xcpdIti55Request An array of patient discovery transaction requests.
+   * @param xcpdIti55Request[].id Unique ID for the request.
+   * @param xcpdIti55Request[].cxId The customer ID.
+   * @param xcpdIti55Request[].timestamp The timestamp of the request.
+   * @param xcpdIti55Request[].xcpdGateways The OIDs and XCPD ITI-55 URLs of each organization to make a request to.
+   * @param xcpdIti55Request[].principalCareProviderNPIs The list of NPIs of the practitioners associated with the patient.
+   * @param xcpdIti55Request[].samlAttributes The SAML attributes of the org making the request.
+   * @param xcpdIti55Request[].patient The patient data in FHIR R4 format.
+   *
+   * @returns 202 Accepted if the request was successful.
+   */
```

Comment:
For the params above would it suffice to just have the `@param xcpdIti55Request An array of patient discovery transaction requests.` or does it make sense to elaborate on each item in the object. It can end being pretty long if we go even further with the nesting